### PR TITLE
autoscaling policies fix and filter 

### PR DESCRIPTION
**Pull request by ashleyschuett**
### Description

Add a filter for autoscaling policies to only see policies associated with a specified node pool. Fixed a bug that was printing a null pointer exception if an asp was not associated with a policy. 

 #### What does this pull request accomplish?

 #### What issue(s) does this fix?

 #### Additional Considerations

I don't know if this should be a separate command and then find the node pool scope, or if it was better to do it as a filter. 

It seemed strange to be to have a separate `get node-pool-autoscaling-policies` command so I did it as a filter, but am fine with changing it. 

 ### Testing

```
$ csctl --cluster 71d8a0c5-73d6-4e3e-ba28-fd46d9bf03f7 get autoscaling-policy
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1426679]

goroutine 1 [running]:
github.com/containership/csctl/resource.(*AutoscalingPolicies).Table(0xc0005382a0, 0x1727d00, 0xc0000b6008, 0x0, 0x0)
	/Users/ashley/go/src/github.com/containership/csctl/resource/autoscaling_policies.go:55 +0x149
github.com/containership/csctl/cmd.outputResponse(0x172e0a0, 0xc0005382a0)
	/Users/ashley/go/src/github.com/containership/csctl/cmd/get.go:24 +0x60
github.com/containership/csctl/cmd.glob..func12(0x1b7fa00, 0xc00052e140, 0x0, 0x2, 0x0, 0x0)
	/Users/ashley/go/src/github.com/containership/csctl/cmd/get_autoscaling_policy.go:44 +0x237
github.com/containership/csctl/vendor/github.com/spf13/cobra.(*Command).execute(0x1b7fa00, 0xc00052e0e0, 0x2, 0x2, 0x1b7fa00, 0xc00052e0e0)
	/Users/ashley/go/src/github.com/containership/csctl/vendor/github.com/spf13/cobra/command.go:762 +0x473
github.com/containership/csctl/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x1b81420, 0xc0004a5f88, 0x1007500, 0xc000090058)
	/Users/ashley/go/src/github.com/containership/csctl/vendor/github.com/spf13/cobra/command.go:852 +0x2fd
github.com/containership/csctl/vendor/github.com/spf13/cobra.(*Command).Execute(0x1b81420, 0x0, 0x0)
	/Users/ashley/go/src/github.com/containership/csctl/vendor/github.com/spf13/cobra/command.go:800 +0x2b
github.com/containership/csctl/cmd.Execute()
	/Users/ashley/go/src/github.com/containership/csctl/cmd/root.go:139 +0x2d
main.main()
	/Users/ashley/go/src/github.com/containership/csctl/main.go:6 +0x20
```

```
$ csctl --cluster 71d8a0c5-73d6-4e3e-ba28-fd46d9bf03f7 get autoscaling-policy
NAME	ID					METRICS_BACKEND				METRIC				SCALE_UP	SCALE_DOWN	POLL_INTERVAL	SAMPLE_PERIOD
new	82ca561c-fd6d-440e-a4cd-9976ec3416ef	6515b40b-46d2-4635-a6f2-56a8640b9bc7	cpu_percent_utilization						15s		300s
```

```
$ csctl --cluster 71d8a0c5-73d6-4e3e-ba28-fd46d9bf03f7 --node-pool 53e62d0c-7daf-4be5-8592-f1ce3b04584c get autoscaling-policy
NAME	ID					METRICS_BACKEND				METRIC				SCALE_UP	SCALE_DOWN	POLL_INTERVAL	SAMPLE_PERIOD
new	82ca561c-fd6d-440e-a4cd-9976ec3416ef	6515b40b-46d2-4635-a6f2-56a8640b9bc7	cpu_percent_utilization
```

```
$ csctl --cluster 71d8a0c5-73d6-4e3e-ba28-fd46d9bf03f7 --node-pool 43e62d0c-7daf-4be5-8592-f1ce3b04584c get autoscaling-policy
Error: server responded with status 404: {"message":"Not Found"}
```

 #### Setup

`make install`

 #### Instructions

 ### Dependencies

### CODE REVIEW

File: cloud/provision/autoscaling_policy.go [hunk 1]
```diff
@@ -65,3 +66,11 @@ func (c *autoscalingPolicies) List() ([]types.AutoscalingPolicy, error) {
 	out := make([]types.AutoscalingPolicy, 0)
 	return out, c.client.Get(path, &out)
 }
+
+// List lists node pool autoscaling policies
```
Comment (nicktate) [inline cloud/provision/autoscaling_policy.go]: ListNodePoolASP lists...

**Comment by ashleyschuett** [comment]
I also don't love the API for this. I have a few different thoughts below, including the current one. 

`cloud.Provision().AutoscalingPolicies(env.OrganizationID(), env.ClusterID()). ListNodePoolASP(nodeepoolid)`

`cloud.Provision().AutoscalingPolicies(env.OrganizationID(), env.ClusterID()).ListForNodePool(nodeepoolid)`

`cloud.Provision().AutoscalingPolicies(env.OrganizationID(), env.ClusterID()).ListByNodePool(nodeepoolid)`

`cloud.Provision().AutoscalingPolicies(env.OrganizationID(), env.ClusterID()).NodePool(nodeepoolid)`

**Comment by nicktate** [comment]
Only difference in those is naming though right? Not really a different API structure. In terms of semantics I lean towards:

1. `ListByNodePool`

**Comment by ashleyschuett** [comment]
Correct. I was liking `ListByNodePool` but then I was thinking I would expect it to return autoscaling groups grouped by node pool, and returning multiple node pools, but this only returns autoscaling groups from one node pool

File: resource/autoscaling_policies.go [hunk 2]
```diff
@@ -50,22 +50,26 @@ func (p *AutoscalingPolicies) columns() []string {
...
 		var upPercent string
-		up := asp.Policy.ScaleUp
-		if *up.AdjustmentType == "percent" {
-			upPercent = "%"
-		}
-		scaleUpPolicy := fmt.Sprintf("%s %.2f: +%d%s",
-			*up.ComparisonOperator, *up.Threshold, *up.AdjustmentValue, upPercent)
+		if asp.Policy != nil {
+			up := asp.Policy.ScaleUp
+			if *up.AdjustmentType == "percent" {
```
Comment (mattkelly) [inline resource/autoscaling_policies.go]: Can't `up` be nil? Should we check for that so it doesn't explode? Looks like this is my bug :)

**Comment by mattkelly** [comment]
I do agree the API is a bit weird. I personally am in favor of renaming it to either `ListForNodePool` or `ListByNodePool`, since the current name is a bit stuttery.

This is part of a more general design decision I've had in the back of my head. Once you break out of the typical `List`, `Create`, etc interface the APIs feel mismatched and kind of gross.

I think the API would be more ergonomic if we chained interfaces. It's just more effort than feels worth it for now, and I'm not sure how it would work with code generation if we ever get that far.

This ties into the decisions to be made around when we should filter in the client vs. filtering server-side. I personally think that adding filtering server-side would be great in most cases, and then we could probably autogenerate a better builder interface for the client based on query parameters.

...but again, in short, I think renaming and merging is great for now.

File: resource/autoscaling_policies.go (hunk 2 above)
Comment (ashleyschuett) [reply resource/autoscaling_policies.go]: nice catch 👍

File: resource/autoscaling_policies.go [hunk 3]
```diff
@@ -50,22 +50,14 @@ func (p *AutoscalingPolicies) columns() []string {
 func (p *AutoscalingPolicies) Table(w io.Writer) error {
 	table := table.New(w, p.columns())
 
+	var scaleUpPolicy string
+	var scaleDownPolicy string
```
Comment (ashleyschuett) [inline resource/autoscaling_policies.go]: realized these are in the wrong place

File: resource/resource.go [hunk 4]
```diff
@@ -1,5 +1,9 @@
 package resource
 
+const (
+	emptyColState = "<none>"
```
Comment (mattkelly) [inline resource/resource.go]: 👍

**Pull request merged**